### DISCUSSION TITLE
How to delete all the existing frameworks and create new one from scratch.?

### DISCUSSION THREAD

**Discussion by swayangjit** [Sunbird-Knowlg/Community#125] [General Discussions]
Steps to delete all the existing frameworks and create new ones from scratch.

Reference: - https://github.com/orgs/Sunbird-Lern/discussions/106#discussioncomment-5672533

**Comment by pallakartheekreddy** [comment]
@swayangjit Please find steps below

1. To delete all the existing frameworks, please use neo4j cypher query. 
**Ref:**` MATCH (n:domain{IL_UNIQUE_ID:"<framework_id>"}) DETACH DELETE n;`
2. Please find the below link for more details about creating a new framework steps
**Ref:** https://knowlg.sunbird.org/learn/product-and-developer-guide/taxonomy-and-tagging/framework-service/faqs

**Reply by ruchinksinha** [reply]
@pallakartheekreddy @swayangjit I was able to delete all the existing framework using the above process. This was confirmed by the framework list API:

URL: http://10.0.1.7:8080/learning-service/framework/v3/list

**Response:**
{
    "id": "ekstep.learning.framework.list",
    "ver": "1.0",
    "ts": "2023-04-27T08:25:10ZZ",
    "params": {
        "resmsgid": "48ba5100-8bb8-4b12-82f9-f0a1a8414b35",
        "msgid": null,
        "err": null,
        "status": "successful",
        "errmsg": null
    },
    "responseCode": "OK",
    "result": {
        "frameworks": [],
        "count": 0
    }
}

Now when I am trying to create a new frame using the below, its giving an error.

**URL:** https://<domain_name>/api/framework/v1/create
**Request:**
{
    "request": {
        "framework": {
            "name": "NCF Demo",
            "code": "ncfdemo",
            "description": "NCF Demo Framework"
        }
    }
}

**Response:**
{
    "id": "api.framework.create",
    "ver": "1.0",
    "ts": "2023-04-27T08:28:42.271Z",
    "params": {
        "resmsgid": "84a1b2f0-e4d5-11ed-914d-e53990590bdb",
        "msgid": null,
        "status": "failed",
        "err": null,
        "errmsg": null
    },
    "responseCode": "SERVER_ERROR",
    "result": {}
}

Please help in debugging the above error.

**Reply by ruchinksinha** [reply]
@pallakartheekreddy Just an observation. It might help us in debugging the above error during framework create.

To list all the existing framework when I use the endpoint "http://10.0.1.7:8080/learning-service/framework/v3/list", this returns below json.
{
    "id": "ekstep.learning.framework.list",
    "ver": "1.0",
    "ts": "2023-04-27T08:36:09ZZ",
    "params": {
        "resmsgid": "6a7905d8-6afb-44ec-aa80-b07532df00ce",
        "msgid": null,
        "err": null,
        "status": "successful",
        "errmsg": null
    },
    "responseCode": "OK",
    "result": {
        "frameworks": [],
        "count": 0
    }
}

But when I use the endpoint "https://<domain_name>/api/framework/v1/list", I get the below error:
{
    "id": "api.framework.list",
    "ver": "1.0",
    "ts": "2023-04-27T08:38:44.565Z",
    "params": {
        "resmsgid": "eba07850-e4d6-11ed-914d-e53990590bdb",
        "msgid": null,
        "status": "failed",
        "err": null,
        "errmsg": null
    },
    "responseCode": "SERVER_ERROR",
    "result": {}
}

Also the error in the knowledgemw pod for the above API call is below:

[2023-04-27T08:40:04.731] [ERROR] api - {"id":"1b681160-e4d7-11ed-b83b-a5953b6eca1e","msg":"Getting error from ekstep while fetching frameworkList","additionalInfo":{"ekStepReqData":{}},"err":{"err":null,"responseCode":"SERVER_ERROR"}}

**Comment by ruchinksinha** [comment]
@pallakartheekreddy Looking forward for your support on resolving this issue. As suggested by Swayangjit, I checked the taxonomy service for any errors. But there we no errors there. I believe the API call to the URL https://<domain_name>/api/framework/v1/list is not reaching the taxonomy service.
Please help in resolving this issue.

**Reply by maheshkumargangula** [reply]
@ruchinksinha - Please check the `knowledge-mw` microservice and `learning-service` (tomcat VM) logs for analysing the errors.
Cc: @pallakartheekreddy

**Reply by ruchinksinha** [reply]
@maheshkumargangula @pallakartheekreddy Below is the error in knowledgemw.

[2023-05-01T04:40:47.505] [ERROR] api - {"id":"577c3cc0-e7da-11ed-8b83-bfa66e80bc78","msg":"Getting error from ekstep while fetching framework by id ","additionalInfo":{"frameworkId":"state_delhi_k_12","queryString":""},"err":{"err":null,"responseCode":"SERVER_ERROR"}}

There are no errors in the learning-service tomcat VM.
